compiler/compiler.cpp

```diff
@@ -248,14 +248,14 @@ bool compiler_execute(CompilerSettings *settings) {
     >> PassC<InlineDefinesUsagesPass>{}
     >> PassC<PreprocessEq3Pass>{}
     >> PassC<PreprocessExceptions>{}
+    >> PipeC<EarlyOptimizationF>{}
```

Comment:
The optimization stage needs to be moved to allow overloading of functions that will accept callable objects.